### ISSUE TITLE
Possible problems with LibSass

### ISSUE THREAD

**Issue by pepelsbey** [ericam/susy#481]
**tl;dr** After updating grunt-sass I got incorrect `width` & `margin` values for the same code.

After updating from grunt-sass 0.18.1 to [grunt-sass 1.0.0](https://github.com/sindresorhus/grunt-sass/releases/tag/v1.0.0) (which introduced node-sass 3.0.0 and libsass 3.2.0) I’ve found some breaking changes in Susy rendering. I also compared it with Sass 3.4.13 output.

Here’s the [gist with output files](https://gist.github.com/pepelsbey/afef0d482d8cdf63da38):

1. libsass-after.css — LibSass output by grunt-sass 1.0.0
2. libsass-before.css — LibSass output by grunt-sass 0.18.1
3. sass.css — Sass 3.4.13 output by grunt-contrib-sass 0.9.2

Source files: [operasoftware/devopera/styles](https://github.com/operasoftware/devopera/tree/master/styles), root file is [screen.scss](https://github.com/operasoftware/devopera/blob/master/styles/screen.scss)

Here are highlights from the files above (with only affected values)
```css
/* libsass-before.css */
@media  (min-width: 569px) and (max-width: 767px) {
    .header__logo {
      width: 30.7692307692%;}
    .header__menu {
      width: 65.3846153846%;
      margin-left: 34.6153846154%;}
    .header__search {
      width: 100%;} }
/* libsass-after.css */
@media (min-width: 569px) and (max-width: 767px) {
    .header__logo {
      width: 14.6341463415%;}
    .header__menu {
      width: 31.7073170732%;
      margin-left: 17.0731707317%;}
    .header__search {
      width: 48.7804878049%;} }
```
I’ll try to dig a bit deeper & file an issue to LibSass once I have time for a proper investigation. Meanwhile I switch to Sass and giving you a hint: _there’s something wrong happening here_.

**Comment by ericam** [comment]
I mainly need to see the input code in order to track this down. Can you simplify it down to one example code-block that consistently gives you different results?

**Comment by pepelsbey** [comment]
Just like I said, source files are located in [operasoftware/devopera/styles](https://github.com/operasoftware/devopera/tree/master/styles) folder, root file is [screen.scss](https://github.com/operasoftware/devopera/blob/master/styles/screen.scss) which outputs to `screen.css`. I’ll try to simplify this code to a simple test case in the next few days.

**Comment by ericam** [comment]
I can't recreate based on this. I built a smaller [sassmeister](http://sassmeister.com/gist/9633eafc481605f3b1d2) demo based on your source files, but I get the same (proper) output from both libsass 3.2.2 and ruby sass. Let me know if you are able to track down a broken demo for me to look at.

**Comment by pepelsbey** [comment]
I’ve just started to simplify it, but I can still reproduce it by following steps:

- Download [styles.zip](https://www.dropbox.com/s/9yp3j5hz4depggp/styles.zip?dl=0), both susy & breakpoint-sass are included in the package
- Unzip, open, run inside:
```
$ sass screen.scss screen-by-sass.css
$ node-sass screen.scss screen-by-libsass.css
```
Compare resulted files. I still see the difference:
![image](https://cloud.githubusercontent.com/assets/105274/7575012/efcfd528-f831-11e4-8d78-dc001a9d6982.png)

**Comment by pepelsbey** [comment]
Right, I’ve simplified to a single file with `susy` & `breakpoint-sass` imports:
```scss
@import 'breakpoint-sass/stylesheets/breakpoint';
@import 'susy/sass/susy';

$susy:(
	container:1200px,
	math:fluid,
	output:isolate,
	global-box-sizing:border-box,
	use-custom:(
		clearfix:true
		)
	);

$palm:569px;
$lap:768px;
$desk:1024px;

$up-to-palm:max-width ($palm - 1px);
$palm-to-lap:$palm ($lap - 1px);
$up-to-lap:max-width ($lap - 1px);
$lap-and-up:$lap;
$lap-to-desk:$lap ($desk - 1px);
$up-to-desk:max-width ($desk - 1px);
$desk-and-up:$desk;

$layouts:(
	$up-to-palm:2 1/8,
	$palm-to-lap:3 1/8,
	$lap-to-desk:6 1/6,
	$desk-and-up:12 1/4
	);

@mixin layout-breakpoint($query) {
	@include susy-breakpoint($query, map-get($layouts, $query)) {
		@content;
		}
	}

.header {
	@include layout-breakpoint($palm-to-lap) {
		&__logo {
			@include span(1 at 1);
			}
		&__menu {
			@include span(2 at 2);
			}
		&__search {
			@include span(3 at 1 break);
			}
		}
	}

.layout {
	@include layout-breakpoint($palm-to-lap) {
		&__aside {
			@include span(1 at 1);
			}
		&__content {
			@include span(2 at 2);
			}
		}
	}
```
![image](https://cloud.githubusercontent.com/assets/105274/7575419/de328588-f834-11e4-9710-c8f049852eb1.png)

**Comment by pepelsbey** [comment]
Just in case:
```
$ node-sass -v
node-sass	3.0.0	(Wrapper)	[JavaScript]
libsass  	3.2.2	(Sass Compiler)	[C/C++]
$ sass -v
Sass 3.4.13 (Selective Steve)
```

**Comment by ericam** [comment]
I can't recreate this. May have been an issue in early versions of libsass?

http://sassmeister.com/gist/c141d5708fd01a8a87e9
